plugins/techdocs/src/reader/transformers/html/transformer.ts

```diff
@@ -41,13 +45,87 @@ export const useSanitizerTransformer = (): Transformer => {
+        config,
+        'allowedIframeHosts',
+      );
+      // holds all allowable iframe attributes since some attributes  might be custom
+      // and will be stripped at the end even if allowed through the uponSanitizeElement
+      const allIframeAllowedAttributes = ['src'];
 
       DOMPurify.addHook('beforeSanitizeElements', removeUnsafeLinks);
       const tags = ['link', 'meta'];
 
-      if (hosts) {
+      if (iframeConfigs) {
         tags.push('iframe');
+
+        const hosts = [];
+        if (isSimpleIframeConfig(iframeConfigs)) {
+          hosts.push(...iframeConfigs);
+        } else {
+          // by default we need to keep track of 'all' original iframe attributes to allow DOMPurify to handle them normally
+          const baseAllowedAttributes = [
+            'allow',
+            'allowfullscreen',
+            'allowpaymentrequest',
+            'browsingtopics',
+            'credentialless',
+            'csp',
+            'height',
+            'loading',
+            'name',
+            'referrerpolicy',
+            'sandbox',
+            'src',
+            'srcdoc',
+            'width',
+          ];
+
+          for (const iframeConfig of iframeConfigs) {
+            // check that the config is correct
+            const validKeys = ['src', 'allowedAttributes'];
+            for (const key of iframeConfig.keys()) {
+              if (!validKeys.includes(key)) {
+                const valid = validKeys.map(k => `'${k}'`).join(', ');
+                throw new Error(
+                  `Invalid key '${key}' in 'allowedIframeHosts' config, expected one of ${valid}`,
+                );
+              }
+            }
+
+            const host = iframeConfig.getString('src');
+            hosts.push(host);
+
+            const currentIframeAllowedAttributes =
+              iframeConfig.getOptionalStringArray('allowedAttributes') || [];
+            // add to our list of allowed attributes for not stripping at the end
+            allIframeAllowedAttributes.push(...currentIframeAllowedAttributes);
+            const allowedAttributes = baseAllowedAttributes.concat(
+              currentIframeAllowedAttributes,
+            );
+
+            DOMPurify.addHook('uponSanitizeElement', (currNode, data) => {
+              if (data.tagName === 'iframe') {
+                if (isHost(currNode, host)) {
+                  for (const attr of currNode.attributes) {
+                    if (
+                      baseAllowedAttributes.includes(
+                        attr.name.toLocaleLowerCase(),
+                      )
+                    ) {
+                      break;
+                    }
+                    if (
+                      !allowedAttributes.includes(attr.name.toLocaleLowerCase())
+                    ) {
+                      currNode.removeAttribute(attr.name);
+                    }
+                  }
+                }
+              }
+            });
+          }
+        }
```

Comment:
This is looking good so far!  I'm wondering if it might make more sense to move this logic into the `removeUnsafeIframes` function?  It seems like a natural fit, and you could use the existing `data` parameter to check if the attributes are allowed as well. What do you think?